tests/test_integration.py

```diff
@@ -0,0 +1,223 @@
+            ]
+
+            # Reshape
+            x2_saved = []
+            for i in range(len(x2_stimuli0_saved)):
+                x2_saved.append([x2_stimuli0_saved[i], x2_stimuli1_saved[i]])
+            self.assertEqual(x2_saved, x2)
+        elif param_type == "singleStimuli":
+            x1_saved = (
+                self.s.db.get_engine()
+                .execute("SELECT x1 FROM experiment_table")
+                .fetchall()
+            )
+            x1_saved = [item for sublist in x1_saved for item in sublist]
+            self.assertTrue(x1_saved == x1)
+
+            x2_saved = (
+                self.s.db.get_engine()
+                .execute("SELECT x2 FROM experiment_table")
+                .fetchall()
+            )
+            x2_saved = [item for sublist in x2_saved for item in sublist]
+            self.assertTrue(x2_saved == x2)
+
+    def check_outcome(self, outcome_type, outcome):
+        if outcome_type == "multiOutcome":
+            outcome0_saved = (
+                self.s.db.get_engine()
+                .execute("SELECT outcome_0 FROM experiment_table")
+                .fetchall()
+            )
+            outcome1_saved = (
+                self.s.db.get_engine()
+                .execute("SELECT outcome_1 FROM experiment_table")
+                .fetchall()
+            )
+            outcome0_saved = [item for sublist in outcome0_saved for item in sublist]
+            outcome1_saved = [item for sublist in outcome1_saved for item in sublist]
+            outcome_saved = []
+            for i in range(len(outcome0_saved)):
+                outcome_saved.append([[outcome0_saved[i]], [outcome1_saved[i]]])
+            self.assertEqual(outcome_saved, outcome)
+        elif outcome_type == "singleOutcome":
+            outcome_saved = (
+                self.s.db.get_engine()
+                .execute("SELECT outcome FROM experiment_table")
+                .fetchall()
+            )
+            outcome_saved = [item for sublist in outcome_saved for item in sublist]
+            self.assertTrue(outcome_saved == outcome)
+
+    @parameterized.expand(all_tests)
+    def test_experiment(self, param_type, outcome_type):
+        x1 = params[param_type]["x1"]
+        x2 = params[param_type]["x2"]
+        outcome = outcomes[outcome_type]
+
+        with open(f"tests/configs/{param_type}" + ".ini", "r") as f:
+            dummy_config = f.read()
+
+        self.setup_request["message"]["config_str"] = dummy_config
+
+        self.s.versioned_handler(self.setup_request)
+
+        i = 0
+        while not self.s.strat.finished:
+            self.s.unversioned_handler(self.ask_request)
+            self.get_tell(x1[i], x2[i], outcome[i])
+            i = i + 1
+            self.s.unversioned_handler(self.tell_request)
+
+        # Experiment id
+        exp_id = self.s.db.get_master_records()[0].experiment_id
+
+        # Get tables data
+        raw_data = self.s.db.get_raw_for(exp_id)
```

Comment:
I believe all three of these lines (207-209) have variables that are assigned but never used (our internal linter is catching this, not sure why OSS isn't, cc @crasanders in case something in github actions can be improved). What is the intent? If it is just to ensure they work and don't throw, you can do: 
```
try: 
    _ = self.s.db.get_raw_for(exp_id)
except Exception as e: # ideally a more specific exception though 
    self.fail(f"self.s.db.get_raw_for failed with exception {e}") #
or whatever error message you want to fail with. 
```
(via https://stackoverflow.com/questions/4319825/python-unittest-opposite-of-assertraises)

Or if you meant additional checks here, you should add them.

Reply:
Oh sorry! I considered comparing them with the values saved in the experiment table. Still, in the end, I checked the values in the experiment table with the original values from the dictionaries!

I'll remove those variables! ✅